### ISSUE TITLE
validateRowInProgress throws TransformError with decode: false

### ISSUE THREAD

**Issue by rochoa** [googleapis/nodejs-bigtable#394]
#### Environment details

  - OS: macOS Sierra 10.12.6
  - Node.js version: v8.14.0
  - npm version: 6.4.1
  - `@google-cloud/bigtable` version: 1.0.0

#### Steps to reproduce

  1. Have a table with multiple cells in the same row
  2. Use `table.createReadStream` with `decode=false` option.
  3. You get a `TransformError: A commit is required between row keys (...)`.

Example:
```js
const options = {
    prefix: 'test',
    filter: [
        {
            column: ['a', 'b']
        },
        {
            column: {
                cellLimit: 1
            }
        }
    ],
    decode: false
}

table.createReadStream(options)
    .on('error', err => {
        console.error(err);
    })
    .on('data', row => {
        console.log(row.id.toString(), row.data)
    })
    .on('end', () => {
        console.log('end');
    });
```

As the key is not being decoded, it will compare two different buffer instances and it will throw at:

https://github.com/googleapis/nodejs-bigtable/blob/ad95d12a5312fe25a1193b9c49ec80fd3e02b0b6/src/chunktransformer.js#L235-L248

**Comment by zrosenbauer** [comment]
I took a look at this and I'm assuming its due to:

src/mutation.js:70
```javascript
    if (userOptions.decode === false) {
      return buf;
    }
```
because (I could be misunderstanding the logic here, I didn't run it locally I just stepped through the code) `newRowKey !== row.key` will always be true because the buffer is being returned instead of the parsed buffer, i.e. a string.

To fix this:

```javascript
      if (
        newRowKey &&
        chunk.rowKey &&
        newRowKey.length !== 0 &&
        newRowKey !== row.key &&
        this.options.decode === true // prevent decode: false from throwing an error
      ) {
```

I'd be happy to open a PR if this is on the right track.

**Comment by rochoa** [comment]
I took a different approach, which is comparing the toString() value:
https://github.com/googleapis/nodejs-bigtable/blob/8b745f18593ead23950387a763134fc9859acc4d/src/chunktransformer.js#L235-L241

That way, you still validate that the keys are committed between row keys.

**Comment by zrosenbauer** [comment]
@rochoa looking at this, is there the possibility row.key will be null? if so I can update my PR, add some guard rails on this to allow that check.

**Comment by zrosenbauer** [comment]
If `row.key` is null or undefined and `row.key.toString()` is called, it will error. I'd like to believe running the check is still necessary (as its a validation method). 

Yea that makes sense for the short circuit, but after looking at this more I feel like shortcircuiting this is undesirable because it seems this check, `validateRowInProgress `, is necessary. Removing this could cause more issues downstream.

**Comment by rochoa** [comment]
> Yea that makes sense for the short circuit, but after looking at this more I feel like shortcircuiting this is undesirable because it seems this check, `validateRowInProgress `, is necessary. Removing this could cause more issues downstream.

My thinking was similar to this, that's why I went with the `row.key && newRowKey.toString() !== row.key.toString()` approach in my fork.

**Comment by zrosenbauer** [comment]
Got ya, I think we're on the same page. In PR I opened (#403) that's what I did but I also added guard rails around `row.key` because if key is null or undefined, calling toString() will throw an error (If this is the desired behavior I think we should check that prior to running the validation and throw an error with a better message). When `this.reset` is called it resets `this.row = {}`. I see multiple checks for `typeof this.row.key !== 'undefined'` so my assumption is this value is not guaranteed to exist.

Let me know if this makes sense or if this value is guaranteed to exist due to some logic flow I missed elsewhere.
